### PR TITLE
Fix duplicate import javax.valid.Valid in resteasy generator

### PR DESCRIPTION
**Pull request by hamburml**
Fixes https://github.com/OpenAPITools/openapi-generator/issues/17069

The resteasy generator generates

> import javax.validation.Valid;

twice when useBeanValidation is enabled.
It is already added here https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/JavaJaxRS/resteasy/model.mustache#L12

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.6.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by hamburml** [comment]
mention @martin-mfg for review

**Comment by hamburml** [comment]
@wing328 could you pls review? Thx

**Comment by wing328** [comment]
thanks for the PR but I couldn't repeat the issue with the latest master using the follow command/test:

```
java -jar modules/openapi-generator-cli/target/openapi-generator-cli.jar generate -g jaxrs-resteasy -i modules/openapi-generator/src/test/resources/3_0/petstore.yaml -o /tmp/resteasy --additional-properties useBeanValidation=true
```

If I add `useJakartaEe=true` but then the output won't compile.

Can you share with me a way to reproduce the issue?

**Comment by hamburml** [comment]
Sure! I have a reproducer here

https://github.com/hamburml/quarkus-openapi-generator

Just clone main, run mvn package and check the Order.java or Pet.java class. You should see one jakarta and one javax import.
image

I am sorry, I haven't used the cli command, always the maven plugin. I tried your command but I do not know where the files are generated.

**Comment by hamburml** [comment]
Sorry, I found it :)
I used

`java -jar modules/openapi-generator-cli/target/openapi-generator-cli.jar generate -g jaxrs-resteasy -i modules/openapi-generator/src/test/resources/3_0/petstore.yaml -o /tmp/resteasy --additional-properties useBeanValidation=true --additional-properties useJakartaEe=true`

And the generated classes have also both imports (one jakarta and one javax).
image

The example doesnt work on my end but it looks like my gradle doesnt work. I normally use maven.

File: modules/openapi-generator/src/main/resources/JavaJaxRS/resteasy/pojo.mustache [hunk 1]
```diff
@@ -1,5 +1,4 @@
 import io.swagger.annotations.*;
-{{#useBeanValidation}}import javax.validation.Valid;{{/useBeanValidation}}
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/JavaJaxRS/resteasy/pojo.mustache:2]: what if  only `useBeanValidation` is set to true but `useJakartaEe` is set to false (default)? 

what about using the following instead?

```
{{^useJakartaEe}}
{{#useBeanValidation}}
import javax.validation.Valid;
{{/useBeanValidation}}
{{/useJakartaEe}}
```

Comment (wing328) [reply modules/openapi-generator/src/main/resources/JavaJaxRS/resteasy/pojo.mustache:2]: please ignore. I think I found the issue

**Comment by wing328** [comment]
tested locally to confirm there's n duplicate import of javax.valid.Valid

**Comment by hamburml** [comment]
Thanks <3 I am not that versed with gradle but I could try to fix the example. The current example would always only work with javax and never with jakarta annotations because it forces java ee 8.

**Comment by wing328** [comment]
> javax and never with jakarta annotations because it forces java ee 8.

Shall we set the source and target to JDK 11 if useJakrataEe is enabled?

**Comment by wing328** [comment]
just tried it. no luck

seems like something is fundamentally wrong, e.g. 

```
[ERROR] /C:/Users/wing3/AppData/Local/Temp/resteasy234/src/main/java/org/openapitools/api/impl/UserApiServiceImpl.java:[15,34] package jakarta.enterprise.context does not exist
```

missing import in pom.xml?

**Comment by hamburml** [comment]
> Shall we set the source and target to JDK 11 if useJakrataEe is enabled?

Yeah, I think would be good. Jakarta EE 10 needs JDK 11 minimum.

**Comment by wing328** [comment]
https://mvnrepository.com/artifact/jakarta.servlet/jakarta.servlet-api/6.1.0

that's the latest version but we're still using 4.x

**Comment by hamburml** [comment]
> seems like something is fundamentally wrong, e.g.

In the build.gradle of the generated example you need
`implementation 'jakarta.platform:jakarta.jakartaee-api:10.0.0'` instead of `javax:javaee-api:8.0`

And I think some other dependencies are outdated.

**Comment by wing328** [comment]
I ran `gradle build` but no luck either:
```
$ gradle build
Starting a Gradle Daemon (subsequent builds will be faster)

FAILURE: Build failed with an exception.

* Where:
Build file 'C:\Users\wing3\AppData\Local\Temp\resteasy234\build.gradle' line: 20

* What went wrong:
A problem occurred evaluating root project 'openapi-jaxrs-resteasy-server'.
> Could not find method compile() for arguments [io.swagger:swagger-annotations:1.5.22] on object of type org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyHandler.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

```

**Comment by wing328** [comment]
> And I think some other dependencies are outdated.

I wonder if you can help file a PR to update these dependencies 🙏

**Comment by hamburml** [comment]
> Could not find method compile() for arguments [io.swagger:swagger-annotations:1.5.22] on object of type org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyHandler.

Had the same. Gradle v7 removed compile command. It is now implementation.

**Comment by hamburml** [comment]
I will try :)

**Comment by wing328** [comment]
> kinda strange that the example has a pom.xml and a gradle file.

The reason is that some users prefer pom/maven while others prefer gradle.

one can manually delete the one they don't need or ignore it in .openapi-generator-ignore: https://github.com/openapitools/openapi-generator/blob/master/docs/customization.md#ignore-file-format

**Comment by wing328** [comment]
to reproduce the gradle issue, please run
```
java -jar modules/openapi-generator-cli/target/openapi-generator-cli.jar generate -g jaxrs-resteasy -i modules/openapi-generator/src/test/resources/3_0/petstore.yaml -o /tmp/resteasy234 --additional-properties useBeanValidation=false --additional-properties useJakartaEe=true

**Pull request merged**